web/src/main/java/org/cbioportal/web/util/DataBinner.java

```diff
@@ -107,23 +99,31 @@ public DataBinner(DataBinHelper dataBinHelper,
         
         return clinicalDataBins;
     }
-
-    public List<DataBin> calculateClinicalDataBins(String attributeId,
-                                                   List<ClinicalData> clinicalData,
-                                                   List<String> ids)
-    {
-        return calculateClinicalDataBins(attributeId, clinicalData, ids, false);
-    }
     
-    public List<DataBin> calculateClinicalDataBins(String attributeId, 
+    public List<DataBin> calculateClinicalDataBins(ClinicalDataBinFilter clinicalDataBinFilter, 
                                                    List<ClinicalData> clinicalData, 
-                                                   List<String> ids,
-                                                   Boolean disableLogScale)
+                                                   List<String> ids)
     {
+        clinicalDataBinFilter.validate();
+
+        String attributeId = clinicalDataBinFilter.getAttributeId();
+        boolean numericalOnly = false;
+      
+        if (!clinicalDataBinFilter.getRange().upperEndpoint().equals(Double.POSITIVE_INFINITY)) {
```

Comment:
Use .all() and hasUpperBound to determine. Double.POSITIVE_INFINITY may vary base on the platform.